evm/src/fixed_recursive_verifier.rs

```diff
@@ -804,12 +940,26 @@ where
             block_inputs
                 .set_proof_with_pis_target(&self.block.parent_block_proof, parent_block_proof);
         } else {
+            // Initialize state_root_after and the block number for correct connection between blocks.
+            let state_trie_root_keys = 24..32;
+            let block_number_key = TrieRootsTarget::SIZE * 2 + 6;
+            let mut nonzero_pis = HashMap::new();
+            for (key, &value) in state_trie_root_keys.zip_eq(&u256_limbs::<F>(
+                public_values.trie_roots_before.state_root.into_uint(),
+            )) {
+                nonzero_pis.insert(key, value);
+            }
+            nonzero_pis.insert(
+                block_number_key,
+                F::from_canonical_usize(public_values.block_metadata.block_number.as_usize())
```

Comment:
Just to make sure I'm not missing anything, these public values will only be verified for the initial block proof of the block chain. So in this case `block_number` will be zero, right? Therefore, this should just be `F::NEG_ONE`.